### PR TITLE
Replacing flag val, test repairs on method calls on Unknown

### PR DESCRIPTION
**Pull request by apanczyk**

### CODE REVIEW

**Comment by apanczyk** [comment]
Works suspended until merge from https://github.com/TouK/nussknacker/pull/1909 to staging.

**Comment by dswiecki** [comment]
@apanczyk is it still a draft ?

**Comment by apanczyk** [comment]
> @apanczyk is it still a draft ?

@dswiecki Pull is still postponed, until changes from @maciej-brzezinski that will replace Unknown classes in tests with real classes. Part of pull request without tests, should be ready to implement, I'm not sure how should I mark it then.

File: engine/interpreter/src/test/scala/pl/touk/nussknacker/engine/spel/SpelExpressionSpec.scala [hunk 1]
```diff
@@ -76,35 +76,35 @@ class SpelExpressionSpec extends FunSuite with Matchers with EitherValues {
...
-    parse(expr, validationCtx, Map.empty, flavour, strictMethodsChecking = false, staticMethodInvocationsChecking = true, disableMethodExecutionForUnknown = false,
+    parse(expr, validationCtx, Map.empty, flavour, strictMethodsChecking = false, staticMethodInvocationsChecking = true, methodExecutionForUnknownAllowed = false,
       dynamicPropertyAccessAllowed = false)
   }
 
   private def parse[T: TypeTag](expr: String, context: Context = ctx, flavour: Flavour = Standard): ValidatedNel[ExpressionParseError, TypedExpression] = {
     val validationCtx = ValidationContext(
       context.variables.mapValuesNow(Typed.fromInstance))
-    parse(expr, validationCtx, Map.empty, flavour, strictMethodsChecking = true, staticMethodInvocationsChecking = true, disableMethodExecutionForUnknown = false,
+    parse(expr, validationCtx, Map.empty, flavour, strictMethodsChecking = true, staticMethodInvocationsChecking = true, methodExecutionForUnknownAllowed = true,
```
Comment (mproch) [inline engine/interpreter/src/test/scala/pl/touk/nussknacker/engine/spel/SpelExpressionSpec.scala:92]: Let's change the flag only in failing tests?

File: engine/flink/process/src/test/scala/pl/touk/nussknacker/engine/process/helpers/ProcessTestHelpers.scala [hunk 2]
```diff
@@ -115,7 +115,7 @@ object ProcessTestHelpers {
         "processHelper" -> WithCategories(ProcessHelper),
         "enum" -> WithCategories(DictInstance(dictId, dictDef)),
         "typedMap" -> WithCategories(TypedMap(Map("aField" -> "123"))))
-      ExpressionConfig(globalProcessVariables, List.empty, List.empty, dictionaries = Map(dictId -> WithCategories(dictDef)))
+      ExpressionConfig(globalProcessVariables, List.empty, List.empty, dictionaries = Map(dictId -> WithCategories(dictDef)), methodExecutionForUnknownAllowed = true)
```
Comment (mproch) [inline engine/flink/process/src/test/scala/pl/touk/nussknacker/engine/process/helpers/ProcessTestHelpers.scala:118]: Let's change the flag only in failing tests?

File: engine/interpreter/src/test/scala/pl/touk/nussknacker/engine/spel/SpelExpressionSpec.scala (hunk 1 above)
Comment (apanczyk) [reply engine/interpreter/src/test/scala/pl/touk/nussknacker/engine/spel/SpelExpressionSpec.scala]: Ok, already changed that. Hope it's improved.

**Review by mproch — APPROVED** [review]

**Pull request merged**